internal/blockprocessor/committer.go

```diff
@@ -360,19 +373,42 @@ func constructDBEntriesForDBAdminTx(tx *types.DBAdministrationTx, version *types
 			value = v
 		}
 
-		db := &worldstate.KVWithMetadata{
+		updateDBIndex := &worldstate.KVWithMetadata{
 			Key:   dbName,
 			Value: value,
 			Metadata: &types.Metadata{
 				Version: version,
 			},
 		}
-		indexForExistingDBs = append(indexForExistingDBs, db)
+		indexForExistingDBs = append(indexForExistingDBs, updateDBIndex)
+
+		// If the index definition is empty, we need to delete all index entries
+		// associated with that. This is simply done by deleteing the index DB
+		if dbIndex == nil || dbIndex.GetAttributeAndType() == nil {
+			if db.Exist(stateindex.IndexDBPrefix + dbName) {
+				toDeleteDBs = append(toDeleteDBs, stateindex.IndexDBPrefix+dbName)
+			}
+			continue
+		}
+
+		if db.Exist(stateindex.IndexDBPrefix + dbName) {
+			continue
+		}
+
+		// Only when the index definition is provided for the existing DB for the
+		// first time, we will reach here
+		indexDB := &worldstate.KVWithMetadata{
+			Key: stateindex.IndexDBPrefix + dbName,
+			Metadata: &types.Metadata{
+				Version: version,
+			},
+		}
+		indexForExistingDBs = append(indexForExistingDBs, indexDB)
```

Comment:
This loop is pretty difficult to read. I suggest to create to booleans that capture the two conditions:
`indexIsDefined := dbIndex != nil && dbIndex.GetAttributeAndType() != nil`
and
`indexAlreadyExists := db.Exist(stateindex.IndexDBPrefix + dbName)`
and then use them and their converse in switch or if-else statements, that will make it clear what you are trying to achieve and make it easier to read and instantly grasp what happens in each case (there are 4, but maybe on is degenerate?).

Reply:
Agree. Hope, I have simplified it.